src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,43 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_STORAGE_CAPACITY = 10;
+    private K[] keys;
+    private V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_STORAGE_CAPACITY];
+        values = (V[]) new Object[MAX_STORAGE_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if ((keys[i] == null && key == null) || (keys[i] != null && keys[i].equals(key))) {
```

Comment:
```suggestion
            if (keys[i] == null && key == null || keys[i] != null && keys[i].equals(key)) {
```
redundant brackets. Read about logical operator priority